ple/games/tetris_engine.py

```diff
@@ -0,0 +1,518 @@
+                     '..OO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '.OO..',
+                     '.O...',
+                     '.....',
+                     '.....']]
+
+I_SHAPE_TEMPLATE = [['..O..',
+                     '..O..',
+                     '..O..',
+                     '..O..',
+                     '.....'],
+                    ['OOOO.',
+                     '.....',
+                     '.....',
+                     '.....',
+                     '.....']]
+
+O_SHAPE_TEMPLATE = [['.OO..',
+                     '.OO..',
+                     '.....',
+                     '.....',
+                     '.....']]
+
+J_SHAPE_TEMPLATE = [['.O...',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..OO.',
+                     '..O..',
+                     '..O..',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '...O.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '.OO..',
+                     '.....',
+                     '.....']]
+
+L_SHAPE_TEMPLATE = [['...O.',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '..OO.',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '.O...',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['.OO..',
+                     '..O..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+T_SHAPE_TEMPLATE = [['..O..',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..OO.',
+                     '..O..',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '..O..',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '.OO..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+PIECES = {'S': S_SHAPE_TEMPLATE,
+          'Z': Z_SHAPE_TEMPLATE,
+          'J': J_SHAPE_TEMPLATE,
+          'L': L_SHAPE_TEMPLATE,
+          'I': I_SHAPE_TEMPLATE,
+          'O': O_SHAPE_TEMPLATE,
+          'T': T_SHAPE_TEMPLATE}
+
+
+class GameState:
+    def __init__(self):
+        global FPSCLOCK, DISPLAYSURF, BASICFONT, BIGFONT
```

Comment:
please adjust this to not be global. seems like you can assign it to the GameState class with self.FPSCLOCK etc.